### PR TITLE
Add docs for FuzzyTargetting and NoPenaltyTargetting

### PR DESCRIPTION
**Pull request by Sollace**
What's this? A _pull request_!?

Mainly adding documentation and changed two method names to better describe what they do.

`NoPenaltyTargetting#find` -> `NoPenaltyTargetting#findTo`
`NoPenaltyTargetting#find` -> `NoPenaltyTargetting#findFrom`

These both find positions leading to and away from a given start/end position.

`findFrom` for instance is used in (and only in) `FleeEntityGoal` (circa 1.17):

```
Vec3d vec3d = NoPenaltyTargeting.findFrom(this.mob, 16, 7, this.targetEntity.getPos());
```

### CODE REVIEW

File: mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping [hunk 1]
```diff
@@ -1,38 +1,78 @@
 CLASS net/minecraft/class_5534 net/minecraft/entity/ai/FuzzyTargeting
 	COMMENT Path targeting utilities using fuzzy, or approximated, positions from
 	COMMENT {@link FuzzyPositions}.
+	COMMENT <p>
+	COMMENT Methods in this class can be used to do pathing for an entity to a random position.
+	COMMENT Positions are chosen to stay within range of the entity's chosen {@link net.minecraft.entity.mob.MobEntity#getPositionTarget() position target}
+	COMMENT if applicable, and will be suitably randomised within that contraint.
 	METHOD method_31527 find (Lnet/minecraft/class_1314;II)Lnet/minecraft/class_243;
+		COMMENT Paths to a random reachable position with positive path-finding favourability.
```
Comment (Juuxel) [inline mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping]:
```suggestion
		COMMENT Paths to a random reachable position with positive path-finding favorability.
```
Yarn uses simplified English :pensive:

File: mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping [hunk 2]
```diff
@@ -1,38 +1,78 @@
 CLASS net/minecraft/class_5534 net/minecraft/entity/ai/FuzzyTargeting
 	COMMENT Path targeting utilities using fuzzy, or approximated, positions from
 	COMMENT {@link FuzzyPositions}.
+	COMMENT <p>
+	COMMENT Methods in this class can be used to do pathing for an entity to a random position.
+	COMMENT Positions are chosen to stay within range of the entity's chosen {@link net.minecraft.entity.mob.MobEntity#getPositionTarget() position target}
+	COMMENT if applicable, and will be suitably randomised within that contraint.
```
Comment (Juuxel) [inline mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping]:
```suggestion
	COMMENT if applicable, and will be suitably randomized within that contraint.
```
Same here.

File: mappings/net/minecraft/entity/ai/NoPenaltyTargeting.mapping [hunk 3]
```diff
@@ -2,20 +2,41 @@ CLASS net/minecraft/class_5532 net/minecraft/entity/ai/NoPenaltyTargeting
...
+			COMMENT the entity doing the pathing
 		ARG 1 horizontalRange
+			COMMENT the horizontal pathing range. This is how far the point can be from the entity's starting position on the X or Z range
 		ARG 2 verticalRange
-	METHOD method_31511 find (Lnet/minecraft/class_1314;IILnet/minecraft/class_243;)Lnet/minecraft/class_243;
+			COMMENT the vertical pathing range. This is how far the point can be from the entity's starting position on the Y range
+	METHOD method_31511 findFrom (Lnet/minecraft/class_1314;IILnet/minecraft/class_243;)Lnet/minecraft/class_243;
+		COMMENT Paths to a position leading away from a given starting point.
+		COMMENT
+		COMMENT @return The chosen end position or null if no valid positions could be found.
```
Comment (Juuxel) [inline mappings/net/minecraft/entity/ai/NoPenaltyTargeting.mapping]:
```suggestion
		COMMENT @return the chosen end position or null if no valid positions could be found
```
The same style rules apply to `@return` and `@param` (although not checked by Filament for returns)

File: mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping [hunk 4]
```diff
@@ -1,38 +1,78 @@
 CLASS net/minecraft/class_5534 net/minecraft/entity/ai/FuzzyTargeting
 	COMMENT Path targeting utilities using fuzzy, or approximated, positions from
 	COMMENT {@link FuzzyPositions}.
+	COMMENT <p>
+	COMMENT Methods in this class can be used to do pathing for an entity to a random position.
+	COMMENT Positions are chosen to stay within range of the entity's chosen {@link net.minecraft.entity.mob.MobEntity#getPositionTarget() position target}
```
Comment (Juuxel) [inline mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping]:
```suggestion
	COMMENT Positions are chosen to stay within range of the entity's chosen {@linkplain net.minecraft.entity.mob.MobEntity#getPositionTarget() position target}
```

File: mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping [hunk 5]
```diff
@@ -1,38 +1,78 @@
...
 		ARG 0 entity
+			COMMENT the entity doing the pathing
 		ARG 1 horizontalRange
+			COMMENT the horizontal pathing range. This is how far the point can be from the entity's starting position on the X or Z range
 		ARG 2 verticalRange
+			COMMENT the vertical pathing range. This is how far the point can be from the entity's starting position on the Y range
 	METHOD method_31528 findTo (Lnet/minecraft/class_1314;IILnet/minecraft/class_243;)Lnet/minecraft/class_243;
+		COMMENT Paths to a random reachable position leading towards a given end-point.
+		COMMENT
+		COMMENT @return The chosen position or null if none could be found.
```
Comment (Juuxel) [inline mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping]:
```suggestion
		COMMENT @return the chosen position or null if none could be found
```

File: mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping [hunk 6]
```diff
@@ -1,38 +1,78 @@
...
 	METHOD method_31529 findValid (Lnet/minecraft/class_1314;IILnet/minecraft/class_243;Z)Lnet/minecraft/class_243;
 		ARG 0 entity
 		ARG 1 horizontalRange
 		ARG 2 verticalRange
 		ARG 3 direction
 		ARG 4 posTargetInRange
 	METHOD method_31530 find (Lnet/minecraft/class_1314;IILjava/util/function/ToDoubleFunction;)Lnet/minecraft/class_243;
+		COMMENT Paths to a random reachable position with positive path-finding favourability computed by a given function.
+		COMMENT
+		COMMENT @return The chosen position or null if none could be found.
```
Comment (Juuxel) [inline mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping]:
```suggestion
		COMMENT @return the chosen position or null if none could be found
```

**Review by Juuxel — CHANGES_REQUESTED** [review]
There are still some names not matching the conventions.

Also, avoiding Filament's check for parameter formatting and still having multiple sentences in them is sketchy. I'd format those in some other way (for example: `the ... range (how far the point...)`).

File: mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping (hunk 1 above)
Comment (Sollace) [reply mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping]: Oof. I remembered for "armor" but my default is correct English. Sorry.

File: mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping [hunk 7]
```diff
@@ -1,38 +1,78 @@
 CLASS net/minecraft/class_5534 net/minecraft/entity/ai/FuzzyTargeting
 	COMMENT Path targeting utilities using fuzzy, or approximated, positions from
 	COMMENT {@link FuzzyPositions}.
+	COMMENT <p>
+	COMMENT Methods in this class can be used to do pathing for an entity to a random position.
+	COMMENT Positions are chosen to stay within range of the entity's chosen {@linkplain net.minecraft.entity.mob.MobEntity#getPositionTarget() position target}
+	COMMENT if applicable, and will be suitably randomized within that contraint.
```
Comment (enbrain) [inline mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping]:
```suggestion
	COMMENT if applicable, and will be suitably randomized within that constraint.
```

File: mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping [hunk 8]
```diff
@@ -1,38 +1,78 @@
...
 		ARG 3 end
+			COMMENT the position to path towards
 	METHOD method_31529 findValid (Lnet/minecraft/class_1314;IILnet/minecraft/class_243;Z)Lnet/minecraft/class_243;
 		ARG 0 entity
 		ARG 1 horizontalRange
 		ARG 2 verticalRange
 		ARG 3 direction
 		ARG 4 posTargetInRange
 	METHOD method_31530 find (Lnet/minecraft/class_1314;IILjava/util/function/ToDoubleFunction;)Lnet/minecraft/class_243;
+		COMMENT Paths to a random reachable position with positive path-finding favourability computed by a given function.
```
Comment (enbrain) [inline mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping]:
```suggestion
		COMMENT Paths to a random reachable position with positive path-finding favorability computed by a given function.
```

File: mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping [hunk 9]
```diff
@@ -1,38 +1,78 @@
...
+		COMMENT @return the chosen position or null if none could be found
 		ARG 0 entity
+			COMMENT the entity doing the pathing
 		ARG 1 horizontalRange
+			COMMENT the horizontal pathing range. This is how far the point can be from the entity's starting position on the X or Z range
 		ARG 2 verticalRange
+			COMMENT the vertical pathing range. This is how far the point can be from the entity's starting position on the Y range
 		ARG 3 scorer
-	METHOD method_31532 tryMake (Lnet/minecraft/class_1314;IZLnet/minecraft/class_2338;)Lnet/minecraft/class_2338;
+			COMMENT tunction to compute the path-finding favourability of a candidate position
```
Comment (enbrain) [inline mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping]:
```suggestion
			COMMENT function to compute the path-finding favorability of a candidate position
```

**Review by enbrain — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping [hunk 10]
```diff
@@ -1,38 +1,78 @@
...
+	COMMENT Positions are chosen to stay within range of the entity's chosen {@linkplain net.minecraft.entity.mob.MobEntity#getPositionTarget() position target}
+	COMMENT if applicable, and will be suitably randomized within that constraint.
 	METHOD method_31527 find (Lnet/minecraft/class_1314;II)Lnet/minecraft/class_243;
+		COMMENT Paths to a random reachable position with positive path-finding favorability.
+		COMMENT
+		COMMENT @return chosen position or null if none could be found
 		ARG 0 entity
+			COMMENT the entity doing the pathing
 		ARG 1 horizontalRange
+			COMMENT the horizontal pathing range. This is how far the point can be from the entity's starting position on the X or Z range
```
Comment (enbrain) [inline mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping]:
```suggestion
			COMMENT the horizontal pathing range (how far the point can be from the entity's starting position on the X or Z range)
```

File: mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping [hunk 11]
```diff
@@ -1,38 +1,78 @@
...
 		ARG 2 verticalRange
+			COMMENT the vertical pathing range. This is how far the point can be from the entity's starting position on the Y range
```
Comment (enbrain) [inline mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping]:
```suggestion
			COMMENT the vertical pathing range (how far the point can be from the entity's starting position on the Y range)
```

File: mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping [hunk 12]
```diff
@@ -1,38 +1,78 @@
...
 	METHOD method_31528 findTo (Lnet/minecraft/class_1314;IILnet/minecraft/class_243;)Lnet/minecraft/class_243;
+		COMMENT Paths to a random reachable position leading towards a given end-point.
+		COMMENT
+		COMMENT @return the chosen position or null if none could be found
 		ARG 0 entity
 		ARG 1 horizontalRange
+			COMMENT the horizontal pathing range. This is how far the point can be from the entity's starting position on the X or Z range
```
Comment (enbrain) [inline mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping]:
```suggestion
			COMMENT the horizontal pathing range (how far the point can be from the entity's starting position on the X or Z range)
```

File: mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping [hunk 13]
```diff
@@ -1,38 +1,78 @@
...
 		ARG 2 verticalRange
+			COMMENT the vertical pathing range. This is how far the point can be from the entity's starting position on the Y range
```
Comment (enbrain) [inline mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping]:
```suggestion
			COMMENT the vertical pathing range (how far the point can be from the entity's starting position on the Y range)
```

File: mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping [hunk 14]
```diff
@@ -1,38 +1,78 @@
...
 		ARG 3 direction
 		ARG 4 posTargetInRange
 	METHOD method_31530 find (Lnet/minecraft/class_1314;IILjava/util/function/ToDoubleFunction;)Lnet/minecraft/class_243;
+		COMMENT Paths to a random reachable position with positive path-finding favorability computed by a given function.
+		COMMENT
+		COMMENT @return the chosen position or null if none could be found
 		ARG 0 entity
+			COMMENT the entity doing the pathing
 		ARG 1 horizontalRange
+			COMMENT the horizontal pathing range. This is how far the point can be from the entity's starting position on the X or Z range
```
Comment (enbrain) [inline mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping]:
```suggestion
			COMMENT the horizontal pathing range (how far the point can be from the entity's starting position on the X or Z range)
```

File: mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping [hunk 15]
```diff
@@ -1,38 +1,78 @@
...
 		ARG 2 verticalRange
+			COMMENT the vertical pathing range. This is how far the point can be from the entity's starting position on the Y range
```
Comment (enbrain) [inline mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping]:
```suggestion
			COMMENT the vertical pathing range (how far the point can be from the entity's starting position on the Y range)
```

File: mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping [hunk 16]
```diff
@@ -1,38 +1,78 @@
...
-	METHOD method_31532 tryMake (Lnet/minecraft/class_1314;IZLnet/minecraft/class_2338;)Lnet/minecraft/class_2338;
+			COMMENT function to compute the path-finding favorability of a candidate position
+	METHOD method_31532 towardTarget (Lnet/minecraft/class_1314;IZLnet/minecraft/class_2338;)Lnet/minecraft/class_2338;
+		COMMENT Paths to a random reachable position approaching an entity's chosen {@link net.minecraft.entity.mob.MobEntity#getPositionTarget() position target}.
+		COMMENT
+		COMMENT @return the chosen position or null if none could be found
 		ARG 0 entity
+			COMMENT the entity doing the pathing
 		ARG 1 horizontalRange
+			COMMENT the horizontal pathing range. This is how far the point can be from the entity's starting position on the X or Z range
```
Comment (enbrain) [inline mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping]:
```suggestion
			COMMENT the horizontal pathing range (how far the point can be from the entity's starting position on the X or Z range)
```

File: mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping [hunk 17]
```diff
@@ -1,38 +1,78 @@
...
 		ARG 1 pos
+			COMMENT the candidate position
 	METHOD method_31534 findFrom (Lnet/minecraft/class_1314;IILnet/minecraft/class_243;)Lnet/minecraft/class_243;
+		COMMENT Paths to a random reachable position leading away from a given starting point.
+		COMMENT
+		COMMENT @return the chosen position or null if none could be found
 		ARG 0 entity
+			COMMENT the entity doing the pathing
 		ARG 1 horizontalRange
+			COMMENT the horizontal pathing range. This is how far the point can be from the entity's starting position on the X or Z range
```
Comment (enbrain) [inline mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping]:
```suggestion
			COMMENT the horizontal pathing range (how far the point can be from the entity's starting position on the X or Z range)
```

File: mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping [hunk 18]
```diff
@@ -1,38 +1,78 @@
...
 		ARG 2 verticalRange
+			COMMENT the vertical pathing range. This is how far the point can be from the entity's starting position on the Y range
```
Comment (enbrain) [inline mappings/net/minecraft/entity/ai/FuzzyTargeting.mapping]:
```suggestion
			COMMENT the vertical pathing range (how far the point can be from the entity's starting position on the Y range)
```

File: mappings/net/minecraft/entity/ai/NoPenaltyTargeting.mapping [hunk 19]
```diff
@@ -2,20 +2,41 @@ CLASS net/minecraft/class_5532 net/minecraft/entity/ai/NoPenaltyTargeting
 	COMMENT Similar to {@link FuzzyTargeting}, but the positions this class' utility methods
 	COMMENT find never have pathfinding penalties.
 	METHOD method_31510 find (Lnet/minecraft/class_1314;II)Lnet/minecraft/class_243;
+		COMMENT Paths to a random reachable position with no penalty.
+		COMMENT
+		COMMENT @return the chosen end position or null if no valid positions could be found
 		ARG 0 entity
+			COMMENT the entity doing the pathing
 		ARG 1 horizontalRange
+			COMMENT the horizontal pathing range. This is how far the point can be from the entity's starting position on the X or Z range
```
Comment (enbrain) [inline mappings/net/minecraft/entity/ai/NoPenaltyTargeting.mapping]:
```suggestion
			COMMENT the horizontal pathing range (how far the point can be from the entity's starting position on the X or Z range)
```

File: mappings/net/minecraft/entity/ai/NoPenaltyTargeting.mapping [hunk 20]
```diff
@@ -2,20 +2,41 @@ CLASS net/minecraft/class_5532 net/minecraft/entity/ai/NoPenaltyTargeting
...
 		ARG 2 verticalRange
-	METHOD method_31511 find (Lnet/minecraft/class_1314;IILnet/minecraft/class_243;)Lnet/minecraft/class_243;
+			COMMENT the vertical pathing range. This is how far the point can be from the entity's starting position on the Y range
```
Comment (enbrain) [inline mappings/net/minecraft/entity/ai/NoPenaltyTargeting.mapping]:
```suggestion
			COMMENT the vertical pathing range (how far the point can be from the entity's starting position on the Y range)
```

File: mappings/net/minecraft/entity/ai/NoPenaltyTargeting.mapping [hunk 21]
```diff
@@ -2,20 +2,41 @@ CLASS net/minecraft/class_5532 net/minecraft/entity/ai/NoPenaltyTargeting
...
+	METHOD method_31511 findFrom (Lnet/minecraft/class_1314;IILnet/minecraft/class_243;)Lnet/minecraft/class_243;
+		COMMENT Paths to a position leading away from a given starting point.
+		COMMENT
+		COMMENT @return the chosen end position or null if no valid positions could be found
 		ARG 0 entity
+			COMMENT the entity doing the pathing
 		ARG 1 horizontalRange
+			COMMENT the horizontal pathing range. This is how far the point can be from the entity's starting position on the X or Z range
```
Comment (enbrain) [inline mappings/net/minecraft/entity/ai/NoPenaltyTargeting.mapping]:
```suggestion
			COMMENT the horizontal pathing range (how far the point can be from the entity's starting position on the X or Z range)
```

File: mappings/net/minecraft/entity/ai/NoPenaltyTargeting.mapping [hunk 22]
```diff
@@ -2,20 +2,41 @@ CLASS net/minecraft/class_5532 net/minecraft/entity/ai/NoPenaltyTargeting
...
 		ARG 2 verticalRange
-		ARG 3 direction
-	METHOD method_31512 find (Lnet/minecraft/class_1314;IILnet/minecraft/class_243;D)Lnet/minecraft/class_243;
+			COMMENT the vertical pathing range. This is how far the point can be from the entity's starting position on the Y range
```
Comment (enbrain) [inline mappings/net/minecraft/entity/ai/NoPenaltyTargeting.mapping]:
```suggestion
			COMMENT the vertical pathing range (how far the point can be from the entity's starting position on the Y range)
```

File: mappings/net/minecraft/entity/ai/NoPenaltyTargeting.mapping [hunk 23]
```diff
@@ -2,20 +2,41 @@ CLASS net/minecraft/class_5532 net/minecraft/entity/ai/NoPenaltyTargeting
...
+		ARG 3 start
+			COMMENT the position to path away from
+	METHOD method_31512 findTo (Lnet/minecraft/class_1314;IILnet/minecraft/class_243;D)Lnet/minecraft/class_243;
+		COMMENT Paths to a position leading towards a given end-point.
+		COMMENT
+		COMMENT @return the chosen end position or null if no valid positions could be found
 		ARG 0 entity
+			COMMENT the entity doing the pathing
 		ARG 1 horizontalRange
+			COMMENT the horizontal pathing range. This is how far the point can be from the entity's starting position on the X or Z range
```
Comment (enbrain) [inline mappings/net/minecraft/entity/ai/NoPenaltyTargeting.mapping]:
```suggestion
			COMMENT the horizontal pathing range (how far the point can be from the entity's starting position on the X or Z range)
```

File: mappings/net/minecraft/entity/ai/NoPenaltyTargeting.mapping [hunk 24]
```diff
@@ -2,20 +2,41 @@ CLASS net/minecraft/class_5532 net/minecraft/entity/ai/NoPenaltyTargeting
...
 		ARG 2 verticalRange
+			COMMENT the vertical pathing range. This is how far the point can be from the entity's starting position on the Y range
```
Comment (enbrain) [inline mappings/net/minecraft/entity/ai/NoPenaltyTargeting.mapping]:
```suggestion
			COMMENT the vertical pathing range (how far the point can be from the entity's starting position on the Y range)
```

**Review by enbrain — APPROVED** [review]
Looks good besides Juuxel's point. (you can batch apply suggestions from the "Files changed" tab.)

**Comment by Sollace** [comment]
Alrighty, thanks @enbrain ! I've committed all suggestions.

**Review by enbrain — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**